src/jobtypes.py

```diff
@@ -0,0 +1,121 @@
+import os
+import json
+import logging
+LOG = logging.getLogger(__name__)
+
+
+def _getgroup(name):
+    """
+    It's about to get group for the test job, like crashtest for crashtest-1.
+    """
+    try:
+        group = name.split('-')[0]
+    except:
+        group = ''
+    return group
+
+
+def _getgroupCode(tbplnames, name):
+    """
+    It's about get group code by given testname, like T-e10s for tp5o-e10s.
+    And there has some useless blank in testname like " mochitest-devtools-chrome-3"
+    to strip that blank off, so we can get the code appropriately.
+    """
+    try:
+        code = tbplnames[name.strip()]['group']
+    except:
+        code = ''
+    return code
+
+
+def _getcode(tbplnames, name):
+    """
+    It's about get the code by given testname, it usually related with the group code
+    like '2' for web-platform-tests-2.   And there has some useless blank in testname
+    like " mochitest-devtools-chrome-3" to strip that blank off,
+    so we can get the code appropriately.
+    """
+    try:
+        code = tbplnames[name.strip()]['code']
+    except:
+        code = ''
+    return code
+
+
+class Treecodes:
+    """This class contain all the mapping we need in SETA and make it works"""
+
+    def __init__(self, repo_name='mozilla-inbound'):
+        # default to query all jobs on mozilla-inbound branch
+        self.tbplnames = {}
+        self.jobtypes = []
+        self.jobnames = []
+        with open(os.getcwd() + '/runnablejobs.json') as data:
+            joblist = json.loads(data.read())['results']
+
+        # skipping pgo - We run this infrequent enough that we should have all pgo results tested
+        self.joblist = [job for job in joblist if job['platform_option'] != 'pgo']
+        if len(self.joblist) > 0:
```

Comment:
hmm, I think it's not an error if the length of joblist is equal to 0. At least this error shouldn't be referred in here. I mean, if there is no job in the list but open file is went well, it can only represent there went something wrong when we retrieve json data.